### PR TITLE
Gramine dataset

### PR DESCRIPTION
**Pull request by izimio**
ADD Gramine on TEE option framework for push secret and check secret

### CODE REVIEW

**Comment by PierreJeanjacquot** [comment]
- update the changelog
- update the generated doc => `npm run build` + commit changes

File: test/lib.test.js [hunk 1]
```diff
@@ -3480,6 +3499,22 @@ describe('[dataset]', () => {
       datasetAddress,
     );
     expect(withSecretRes).toBe(true);
+
+    const DatasetDeployedGramine = await deployRandomDataset(iexec);
```
[the remark this answers is not in the document]
Comment (izimio) [reply test/lib.test.js]: f0b1ff6e21c24cdc1aec03a02364fe9864bd5678

[the remark this answers is not in the document]
Comment (izimio) [reply test/lib.test.js]: removed the `.only` just after, sorry for that mistake :(

File: test/lib.test.js [hunk 2]
```diff
@@ -3480,6 +3499,22 @@ describe('[dataset]', () => {
...
+    const datasetAddressGramine = DatasetDeployedGramine.address;
+    await iexec.dataset.pushDatasetSecret(datasetAddressGramine, 'oops', {
+      teeFramework: TEE_FRAMEWORKS.GRAMINE,
+    });
+    const wrogTeeRes = await iexec.dataset.checkDatasetSecretExists(
```
[the remark this answers is not in the document]
Comment (izimio) [reply test/lib.test.js]: acc395bb506a80d3a225867f8649a1e5f46eeb29

File: test/lib.test.js [hunk 3]
```diff
@@ -3480,6 +3499,22 @@ describe('[dataset]', () => {
...
+      datasetAddressGramine,
+      { teeFramework: TEE_FRAMEWORKS.SCONE },
+    );
+    expect(wrogTeeRes).toBe(false);
+    const GoodTeeRes = await iexec.dataset.checkDatasetSecretExists(
```
[the remark this answers is not in the document]
Comment (izimio) [reply test/lib.test.js]: 9259ff7ed72b863e65d42b144e00bb5034c333b5

File: test/lib.test.js [hunk 4]
```diff
@@ -3400,6 +3400,25 @@ describe('[dataset]', () => {
     ).rejects.toThrow(
       Error(`Secret already exists for ${datasetAddress} and can't be updated`),
     );
+    const newDatasetDeployRes = await iexec.dataset.deployDataset({
```
[the remark this answers is not in the document]
Comment (izimio) [reply test/lib.test.js]: 30dc6a0fbdc6b57009afb5f4b82da3ec7672e14c

**Review by PierreJeanjacquot — APPROVED** [review]

**Pull request merged**